onnx/defs/math/defs.cc

```diff
@@ -3660,4 +3660,534 @@ ONNX_OPERATOR_SET_SCHEMA(
+                const int64_t component_ndim = 1;
+                
+                // To support multidimensional DFT the signal_ndim may != 1,
+                // but this attribute is not included on the schema yet
+                const auto signal_ndim = 1;
+
+                // Add 1 dimensions to the rank for the batch size dimension
+                auto real_inputs_ndim = signal_ndim + batch_ndim;
+                // Add 2 dimensions to the rank for the batch size dimension AND component dimension
+                auto complex_inputs_ndim = signal_ndim + batch_ndim + component_ndim;
+
+                auto& input_shape = getInputShape(ctx, 0);
+                ONNX_NAMESPACE::TensorShapeProto result_shape = input_shape;
+
+                auto dim_size = static_cast<int64_t>(input_shape.dim_size());
+                if (dim_size == real_inputs_ndim) {
+                    // For real input: the output is same shape, but with extra dim to represent complex output (real/imaginary)
+                    result_shape.add_dim()->set_dim_value(2);
+                } else if (dim_size == complex_inputs_ndim) {
+                    // For complex input: do nothing
+                } else {
+                  fail_shape_inference(
+                        "The input_shape must [batch_size][n_fft] for real values or [batch_size][n_fft][2] for complex values.")
+                }
+
+                updateOutputShape(ctx, 0, result_shape);
+            }));
+
+static const char* HannWindow_ver13_doc =
+    R"DOC(Generate a Hann window.)DOC";
+
+    ONNX_OPERATOR_SET_SCHEMA(HannWindow,
+        13,
+        OpSchema()
+            .SetDoc(HannWindow_ver13_doc)
+            .Attr(
+                "output_datatype",
+                "The data type of the output tensor. "
+                "Strictly must be one of the values from DataType enum in TensorProto whose values correspond to T2. "
+                "The default value is 1 = FLOAT. ",
+                AttributeProto::INT,
+                static_cast<int64_t>(TensorProto_DataType_FLOAT))
+            .Input(0,
+                   "size",
+                   "A scalar value indicating the length of the window.",
+                   "T1",
+                   OpSchema::Single,
+                   true,
+                   1,
+                   OpSchema::NonDifferentiable)
+            .Output(0,
+                   "output",
+                   "A Hann window with length: size. "
+                   "The output has the shape: [size].",
+                   "T2",
+                   OpSchema::Single,
+                   true,
+                   1,
+                   OpSchema::NonDifferentiable)
+            .TypeConstraint(
+                "T1",
+                {"tensor(int64)"},
+                "Constrain the input size to int64_t.")
+            .TypeConstraint(
+                "T2",
+                {"tensor(float)",
+                 "tensor(float16)",
```

Comment:
Is there a reason for not including bfloat16? Suggest also replacing this with one of the predefined sets, like ```all_numeric_types_with_bfloat()``` : https://github.com/onnx/onnx/blob/5282dd787b2c913110c26f34de7b624cf7ab9c01/onnx/defs/schema.h#L633